### ISSUE TITLE
Worlds won't load because of Datapack errors in master

### ISSUE THREAD

**Issue by SleepyTonia** [ExcessiveAmountsOfZombies/Croptopia#127]
**Version**
Croptopia Fabric 1.16.4 - 1.4.0 (Master)

**Platform**
Fabric

**Describe the issue**
While the game loads, reaches the menu without issue and ModMenu shows that Croptopia indeed was loaded as a mod, whenever you try to open or create a new world, the game shows a screen saying datapacks are broken.

**To Reproduce**
Install a build from the latest master source, start the game and try to open/create a world.

Switching back to Croptopia 1.3.6 fixes this issue.

**Other Mods**
A whole lot… I'm working on a 280+ mods/datapacks modpack ^^;
The most critical things to note in this case however would be:
`fabric-api-0.34.2+1.16.jar`
`fabric-loader-0.11.3-1.16.5`
`Minecraft 1.16.5`

**Additional context**
> [10:38:20] [main/WARN]: Failed to load datapacks, can't proceed with server load
> java.util.concurrent.ExecutionException: net.minecraft.class_151: Non [a-z0-9_.-] character in namespace of location: ${dependant}:butters
> 	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357) ~[?:1.8.0_202]
> 	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895) ~[?:1.8.0_202]
> 	at net.minecraft.class_310.method_29604(class_310.java:1959) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_310.method_29610(class_310.java:1808) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_310.method_29606(class_310.java:1748) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_528$class_4272.method_20174(class_528.java:407) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_528$class_4272.method_20164(class_528.java:317) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_528$class_4272.method_25402(class_528.java:254) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_350.method_25402(class_350.java:345) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_4069.method_25402(class_4069.java:27) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_312.method_1611(class_312.java:92) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_437.method_25412(class_437.java:435) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_312.method_1601(class_312.java:92) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_312.method_22686(class_312.java:162) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_1255.execute(class_1255.java:94) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_312.method_22684(class_312.java:162) ~[intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at org.lwjgl.glfw.GLFWMouseButtonCallbackI.callback(GLFWMouseButtonCallbackI.java:36) [lwjgl-glfw-3.2.2.jar:build 10]
> 	at org.lwjgl.system.JNI.invokeV(Native Method) ~[lwjgl-3.2.2.jar:build 10]
> 	at org.lwjgl.glfw.GLFW.glfwWaitEventsTimeout(GLFW.java:3174) [lwjgl-glfw-3.2.2.jar:build 10]
> 	at com.mojang.blaze3d.systems.RenderSystem.limitDisplayFPS(RenderSystem.java:127) [intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_310.method_1523(class_310.java:1071) [intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.class_310.method_1514(class_310.java:681) [intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at net.minecraft.client.main.Main.main(Main.java:215) [intermediary-fabric-loader-0.11.3-1.16.5.jar:?]
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_202]
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_202]
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_202]
> 	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_202]
> 	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226) [fabric-loader-0.11.3.jar:?]
> 	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:146) [fabric-loader-0.11.3.jar:?]
> 	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28) [fabric-loader-0.11.3.jar:?]

**Comment by ExcessiveAmountsOfZombies** [comment]
Just going to be thorough, you did run the build command with gradlew :fabric:build -Plaunch_mod=c right? Why build it from source if you're just compiling a modpack?

**Comment by BrekiTomasson** [comment]
As a fellow modpack developer, I can chime in with why I enjoy building from source. It's basically to ensure the safety of the code in question, so that nothing malicious or privacy-threatening has been slipped into the compiled JAR file that differs from the code in the repository.

**Comment by SleepyTonia** [comment]
Sorry! This one's entirely my bad, I'd omitted the `-Plaunch_mod=c` part when building. I'm not familiar with this way of building both the fabric and forge versions of a mod. I used the proper build command and it worked. Sorry again!

In my case it's because I've had to apply my own bugfixes several times for various mods, or wanted add/tweak features or alter/remove recipes/loot tables or disable some unwanted features when config options were non-existant. I've specified I was compiling from master because sometimes when I encounter weird crashes from the latest release, some modders will have applied a [bugfix](https://github.com/ExcessiveAmountsOfZombies/Croptopia/commit/4141346ff915c86f65d584e8e776848c49a1d14d) without having published the micro-update yet. I saw that commit and thought I'd give building the mod a try to see if that would work since it did a few times for me.
